OpenRA.Mods.Common/Traits/Render/Hovers.cs

```diff
@@ -43,7 +43,9 @@ public IEnumerable<IRenderable> ModifyRender(Actor self, WorldRenderer wr, IEnum
 			if (self.World.Paused || IsTraitDisabled)
 				return r;
 
-			var visualOffset = !aircraft || self.CenterPosition.Z > 0 ? (int)Math.Abs((self.ActorID + Game.LocalTick) / 5 % 4 - 1) - 1 : 0;
+			var visualOffset = !aircraft || self.World.Map.DistanceAboveTerrain(self.CenterPosition).Length > 0
```

Comment:
There could also be a third way: Since hover mrls in ts will spin to the ground when disabled via emp or ion storm, how about giving hovering vehicles a real altitude when hovering as well (probably in a second trait) and extend the FallsToEarth trait in a way where the actor can survive the crash on a fitting terrain (e.g. get killed on water). So checking for >= ground would still work.